File: smartcityBE/src/main/java/com/smart/service/HotelService.java
package com.smart.service;

import com.smart.model.city.Hotel;
import com.smart.model.city.School;

import java.util.Set;

public interface HotelService {
    public Hotel addHotel(Hotel hotel);
    public Hotel updateHotel(Hotel hotel);
    public Set<Hotel> getHotels();
    public Hotel getHotel(Long hid);
    public void deleteHotel(Long hid);
}

File: CityGuideFE/src/app/services/hotel.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class HotelService {

  constructor(private _http:HttpClient) {}
  
  public hotel(){
    return this._http.get(`${baseUrl}/hotel/`);
  }
  public addHotel(hotel: any){
    return this._http.post(`${baseUrl}/hotel/`, hotel);
  }
  public deleteHotel(hid: any){
    return this._http.delete(`${baseUrl}/hotel/${hid}`);
  }

  public getHotel(hid: any){
    return this._http.get(`${baseUrl}/hotel/${hid}`);
  }

  public updateHotel(hotel: any){
    return this._http.put(`${baseUrl}/hotel/`,hotel);
  }
}
